Python/pystate.c

```diff
@@ -1195,6 +1195,70 @@ _PyThread_CurrentFrames(void)
     return result;
 }
 
+/* The implementation of sys._current_exceptions(). This is intended to be
+   called with the GIL held, as it will be when called via
+   sys._current_frames(). It's possible it would work fine even without the GIL
+   held, but haven't thought enough about that.
+*/
+PyObject *
+_PyThread_CurrentExceptions(void)
+{
+    PyThreadState *tstate = _PyThreadState_GET();
```

Comment:
I would prefer to ensure that the GIL is held. For example, call _Py_EnsureTstateNotNULL(tstate).

Reply:
Done.